### ISSUE TITLE
Add configuration to customize or disable Content Security Policy

### ISSUE THREAD

**Issue by bukharov** [cypress-io/cypress#29596]
### What would you like?

In this file you have a Content Security Policy that governs the test runner (line 12):

https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/util/privileged_channel.ts

We'd like to be able to disable that or add directives to the policy.

### Why is this needed?

When running using ESBuilld it prevents dynamic imports from working. Some of our tests have dynamic imports. Currently everything works fine with Webpack because it bundles everything together. ESBuild does not do that, so those imports stop working. We'd like to use ESBuild instead as the rest of the repository has been migrated to Vite.

**Comment by jennifer-shehane** [comment]
Does the `experimentalCspAllowList` help with your needs? https://docs.cypress.io/guides/references/experiments

**Comment by bukharov** [comment]
@jennifer-shehane no, as far as i understand that is for the application under test, while the one i referenced is for the tests themselves

**Comment by bukharov** [comment]
Basically we get a bunch of these 👇 for each dynamic import in the tests.
<img width="1517" alt="Screenshot 2024-06-03 at 5 17 11 PM" src="https://github.com/cypress-io/cypress/assets/49371/848b882c-787d-4a06-8f9a-3683440dce65">

**Comment by tomatkapitalise** [comment]
Got the same issue as well, my scenario is when trying to use [pdfjs-dist](https://www.npmjs.com/package/pdfjs-dist) which requires you set `workerSrc` which loads a Worker script in the browser context. I'd like to use PDF.js to process a PDF.

**Comment by al-mcnichol** [comment]
I'm looking for answers to the exact same issue and any help will be appreciated
